File: js/reverseTheCase.js
// Given a string, create a function to reverse the case.
// All lower-cased letters should be upper-cased, and vice versa.
// https://edabit.com/challenge/PTvYKkvf7oytHmJFm

const reverseCase = (str) => [...str].map(reverseCharCase).join("");

const reverseCharCase = (char) =>
  char === char.toUpperCase() ? char.toLowerCase() : char.toUpperCase();

module.exports = reverseCase;

File: python/reverse_the_case.py
"""
Reverse the Case

Given a string, create a function to reverse the case. All lower-cased letters should be upper-cased, and vice versa.

https://edabit.com/challenge/o3LWAZtjARJHLtyyb
"""


def reverse_case(txt: str) -> str:
    return "".join(char.swapcase() for char in txt)
